core/metacling/src/TClingGenericInfo.h

```diff
@@ -0,0 +1,46 @@
+/// \file TClingGenericInfo.h
+///
+/// \brief The file contains a base class of TCling*Info classes.
+///
+/// \author Vassil Vassilev <vvasilev@cern.ch>
+///
+/// \date March, 2019
+///
+/*************************************************************************
+ * Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_TClingGenericInfo
+#define ROOT_TClingGenericInfo
+
+#include <string>
+
+namespace clang {
+   class Decl;
+}
+
+class TClingGenericInfo {
+protected:
+   const clang::Decl* fDecl = nullptr;
```

Comment:
Should we consider using a clang::NamedDecl here?

Reply:
We should. I want to do such kinds of refactorings on a later stage.